modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -0,0 +1,789 @@
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/** Partition replication listener. */
+public class PartitionReplicaListener implements ReplicaListener {
+    /** Logger. */
+    private static final IgniteLogger LOG = Loggers.forClass(PartitionReplicaListener.class);
+
+    /** Primary key id. */
+    public static final UUID INDEX_PK_ID = new UUID(0L, 2L);
+
+    /** Scan index id. */
+    public static final UUID INDEX_SCAN_ID = new UUID(0L, 1L);
+
+    /** Table id. */
+    private final UUID tableId;
+
+    /** Versioned partition storage. */
+    private final MvPartitionStorage mvDataStorage;
+
+    /** Raft client. */
+    private final RaftGroupService raftClient;
+
+    /** Lock manager. */
+    private final LockManager lockManager;
+
+    //TODO: https://issues.apache.org/jira/browse/IGNITE-17205 Temporary solution until the implementation of the primary index is done.
+    /** Dummy primary index. */
+    private final ConcurrentHashMap<ByteBuffer, RowId> primaryIndex;
+
+    /**
+     * The constructor.
+     *
+     * @param mvDataStorage Data storage.
+     * @param raftClient    Raft client.
+     * @param lockManager   Lock manager.
+     */
+    public PartitionReplicaListener(
+            MvPartitionStorage mvDataStorage,
+            RaftGroupService raftClient,
+            LockManager lockManager,
+            UUID tableId,
+            ConcurrentHashMap<ByteBuffer, RowId> primaryIndex
+    ) {
+        this.mvDataStorage = mvDataStorage;
+        this.raftClient = raftClient;
+        this.lockManager = lockManager;
+        this.tableId = tableId;
+        this.primaryIndex = primaryIndex;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public CompletableFuture<Object> invoke(ReplicaRequest request) {
+        if (request instanceof ReadWriteSingleRowReplicaRequest) {
+            return processSingleEntryAction((ReadWriteSingleRowReplicaRequest) request);
+        } else if (request instanceof ReadWriteMultiRowReplicaRequest) {
+            return processMultiEntryAction((ReadWriteMultiRowReplicaRequest) request);
+        } else if (request instanceof ReadWriteDualRowReplicaRequest) {
+            return processTwoEntriesAction((ReadWriteDualRowReplicaRequest) request);
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Returns index id of default {@lonk INDEX_SCAN_ID} index that will be used for operation.
+     *
+     * @param indexId Index id or {@code null}.
+     * @return Index id.
+     */
+    private @NotNull UUID indexIdOrDefault(@Nullable UUID indexId) {
+        return indexId != null ? indexId : INDEX_SCAN_ID;
+    }
+
+    /**
+     * Find out a row id by an index.
+     *
+     * @param indexId Index id.
+     * @param key     Key to find.
+     * @return Value or {@code null} if the key does not determine a value.
+     */
+    private RowId valueByUniqueIndex(@NotNull UUID indexId, BinaryRow key) {
```

Comment:
Method name is confusing, it's either valueByKey or rowIdByKey.

Reply:
Renamed.